src/main.rs

```diff
@@ -7,13 +7,13 @@ use std::cell::{Cell, RefCell};
 use std::f32;
 use std::path::PathBuf;
 use std::rc::Rc;
-use std::str::FromStr;
 use std::sync::atomic::{AtomicBool, Ordering};
 use std::sync::{Arc, Mutex};
 use std::time::{Duration, Instant};
 
 use directories::ProjectDirs;
 use lazy_static::lazy_static;
+#[cfg(feature = "networking")]
 use serde::Deserialize;
```

Comment:
I'm assuming that this is for avoiding a warning message about the unused import. I've been thinking how could this be done in a better way because right now this networking condition shows up at quite a lot of places which is not clean in my opinion. The best solution I could think of was to make a separate module for networking. Or to be exact two modules with the same name, like this
```rust 
#[cfg(networking)]
mod networking {
    struct ReleaseInfoJson {}
    pub fn check_for_updates() -> bool {
        // Do the magic
    }
}
#[cfg(not(networking))]
mod networking {
    pub fn check_for_updates() -> bool {
        false
    }
}
```

Otherwise this look really good.

Reply:
Another solution is to move the `ReleaseInfoJson` struct into the check_for_updates function.

Reply:
Yes, that may be even better!

Reply:
Done.